src/tests/core/txs/test_validate_tx.c

```diff
@@ -0,0 +1,289 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include "utxo_to_tx.h"
+#include "crypto.h"
+#include "utxo_pool.h"
+#include "mempool.h"
+#include "minunit.h"
+#include "validate_tx.h"
+
+int tests_run = 0;
+
+mbedtls_ecdsa_context *last_key_pair;
+Transaction *_make_tx() {
+  Transaction *tx;
+  Input *in;
+  Output *out;
+  mbedtls_ecdsa_context *key_pair;
+  mbedtls_ecdsa_context *output_key_pair;
+  tx = malloc(sizeof(Transaction));
+  in = malloc(sizeof(Input));
+  out = malloc(sizeof(Output));
+
+  memset(in, 0, sizeof(Input));
+  key_pair = gen_keys();
+  in->pub_key = &(key_pair->private_Q);
+  in->sig_len = 0; 
+
+  memset(out, 0, sizeof(Output));
+
+  output_key_pair = gen_keys();
+  hash_pub_key(out->public_key_hash, output_key_pair);
+  last_key_pair = output_key_pair;
+  tx->num_inputs = 1;
+  tx->inputs = in;
+  tx->num_outputs = 1;
```

Comment:
Would it be a significant amount of work to do 2 inputs and outputs? I think we might catch bugs that we otherwise won't. If it breaks our testing framework we've written so far then I don't think its worth it, but if its easy we should do it.

Reply:
The hardest part is the signatures, I think we should do it. but after we do a mock, otherwise there will be a lot of weird key copying (or existing helper functions should return a set of keys that were generated to test signatures).

Alternatively, several tests can be run without signatures